tsdb/db.go

```diff
@@ -269,6 +269,12 @@ type DBReadOnly struct {
 	closed  chan struct{}
 }
 
+// Queryable implements a layer for context cancelations over the Querier.
+// Cancelation of any query would lead to rollback and close
```

Comment:
Queries are read-only. What are you trying to do?

Reply:
@brian-brazil  please correct me if I have misunderstood, the issue #5878 asks to implement cancellation in order to avoid issues due to any failure to any higher-level task. Hence, I wrapped the Querier with context cancellation layer which could be stopped at the higher-level if any error arises and closes the db.

Reply:
have I understood the issue correctly?

Reply:
What you propose doesn't cover the bits of the api that matter, opening the DB is not the interesting bit. It's the actual querying/appending that's of interest.

Last we talked about this, we managed to get things fast enough that contexts weren't needed yet and we'll add them once that's not the case.

@jacksontj @krasi-georgiev  FYI

Reply:
I think there were still few places that would benefit a context cancellation, but this would need testing and if context can be avoided by further optimizations than yes we should avoid it. 

Btw I don't see how does this PR can stop a long running query.

Reply:
@krasi-georgiev  I was about to implement content cancellation in the entire package including the appending part intensively, but reading the suggestions, I felt that the importance of content implementation is not of much use as of now. I can resume the implementation if we are in need of this. What do you think? @brian-brazil  @krasi-georgiev @jacksontj

Reply:
yes It would be best to implement this with a case where an optimization doesn't solve the problem.

Reply:
okay @krasi-georgiev , would be moving ahead as suggested and modify the PR accordingly. Thanks

Reply:
would you mind closing this one as by the time this is needed the PR would be very outdated so not worth keeping it open.

Reply:
sure